nimsuggest/nimsuggest.nim

```diff
@@ -1,1164 +1,8 @@
-  result.add ':'
-  result.addInt line
-  result.add ':'
-  result.addInt col
-
-proc replEpc(x: ThreadParams) {.thread.} =
-  var server = newSocket()
-  let port = connectToNextFreePort(server, "localhost")
-  server.listen()
-  echo port
-  stdout.flushFile()
-
-  var client: Socket
-  # Wait for connection
-  accept(server, client)
-  while true:
-    var
-      sizeHex = ""
-      size = 0
-      messageBuffer = ""
-    checkSanity(client, sizeHex, size, messageBuffer)
-    let
-      message = parseSexp($messageBuffer)
-      epcApi = message[0].getSymbol
-    case epcApi
-    of "call":
-      let
-        uid = message[1].getNum
-        cmd = message[2].getSymbol
-        args = message[3]
-
-      when false:
-        x.ideCmd[] = parseIdeCmd(message[2].getSymbol)
-        case x.ideCmd[]
-        of ideSug, ideCon, ideDef, ideUse, ideDus, ideOutline, ideHighlight:
-          setVerbosity(0)
-        else: discard
-      let fullCmd = cmd & " " & args.argsToStr
-      myLog "MSG CMD: " & fullCmd
-      requests.send(fullCmd)
-      toEpc(client, uid)
-    of "methods":
-      returnEpc(client, message[1].getNum, listEpc())
-    of "epc-error":
-      # an unhandled exception forces down the whole process anyway, so we
-      # use 'quit' here instead of 'raise'
-      quit("received epc error: " & $messageBuffer)
-    else:
-      let errMessage = case epcApi
-                       of "return", "return-error":
-                         "no return expected"
-                       else:
-                         "unexpected call: " & epcApi
-      quit errMessage
-
-proc execCmd(cmd: string; graph: ModuleGraph; cachedMsgs: CachedMsgs) =
-  let conf = graph.config
-
-  template sentinel() =
-    # send sentinel for the input reading thread:
-    results.send(Suggest(section: ideNone))
-
-  template toggle(sw) =
-    if sw in conf.globalOptions:
-      excl(conf.globalOptions, sw)
-    else:
-      incl(conf.globalOptions, sw)
-    sentinel()
-    return
-
-  template err() =
-    echo Help
-    sentinel()
-    return
-
-  var opc = ""
-  var i = parseIdent(cmd, opc, 0)
-  case opc.normalize
-  of "sug": conf.ideCmd = ideSug
-  of "con": conf.ideCmd = ideCon
-  of "def": conf.ideCmd = ideDef
-  of "use": conf.ideCmd = ideUse
-  of "dus": conf.ideCmd = ideDus
-  of "mod": conf.ideCmd = ideMod
-  of "chk": conf.ideCmd = ideChk
-  of "highlight": conf.ideCmd = ideHighlight
-  of "outline": conf.ideCmd = ideOutline
-  of "quit":
-    sentinel()
-    quit()
-  of "debug": toggle optIdeDebug
-  of "terse": toggle optIdeTerse
-  of "known": conf.ideCmd = ideKnown
-  of "project": conf.ideCmd = ideProject
-  of "changed": conf.ideCmd = ideChanged
-  of "globalsymbols": conf.ideCmd = ideGlobalSymbols
-  of "declaration": conf.ideCmd = ideDeclaration
-  of "expand": conf.ideCmd = ideExpand
-  of "chkfile": conf.ideCmd = ideChkFile
-  of "recompile": conf.ideCmd = ideRecompile
-  of "type": conf.ideCmd = ideType
-  else: err()
```

Comment:
This is the old re-implimentation of the compilers parsing code. As you can see it is case-insensitive while the compilers is not